plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ManifestsTable.java

```diff
@@ -72,6 +72,7 @@ public ManifestsTable(SchemaTableName tableName, Table icebergTable, Optional<Lo
                         .add(new ColumnMetadata("deleted_data_files_count", INTEGER))
                         .add(new ColumnMetadata("partitions", new ArrayType(RowType.rowType(
                                 RowType.field("contains_null", BOOLEAN),
+                                RowType.field("contains_nan", BOOLEAN),
```

Comment:
same for `$partitions`?

Reply:
Good catch. 
I started looking into how to introduce `nan_count`  for the `PartitionTable`, but this involves a larger refactoring for the `trino-iceberg` connector (writers included). I'll create a different PR for this one

Reply:
https://github.com/trinodb/trino/pull/10709 WIP